ragna/assistants/_ai21labs.py

```diff
@@ -28,7 +28,9 @@ async def answer(
         # See https://docs.ai21.com/reference/j2-chat-api#chat-api-parameters
         # See https://docs.ai21.com/reference/j2-complete-api-ref#api-parameters
         # See https://docs.ai21.com/reference/j2-chat-api#understanding-the-response
-        prompt, sources = (message := messages[-1]).content, message.sources
+        *_, current_message = (
+            message for message in messages if message.role != "system"
+        )
```

Comment:
Likely the same output in most situations, but why not search for the last user message?

```suggestion
        current_prompt = next(m for m in reversed(messages) if m.role == "user")
```

Reply:
Yes, duh. This is what I wanted. Should have been obvious to me :roll_eyes: